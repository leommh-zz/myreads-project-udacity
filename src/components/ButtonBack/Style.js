const style = {
    main: {
        padding: 20,
    },
    input: {
        backgroundColor: '#c8a3ff',
        padding: 20,
    },
    close:{
        backgroundColor: '#c8a3ff',
        marginBottom: 10,
    },
}

export default style;