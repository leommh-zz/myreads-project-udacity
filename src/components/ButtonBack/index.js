import React from 'react';
import { Grid, Button } from '@material-ui/core';
import ArrowBack from '@material-ui/icons/ArrowBack';
import { Link } from 'react-router-dom';
import style from './style';

const ButtonBack = () => (
    <Grid item>
        <Link to="/">
            <Button variant="fab" aria-label="Close" style={style.close}>
                <ArrowBack />
            </Button>
        </Link>
    </Grid>
)

export default ButtonBack;