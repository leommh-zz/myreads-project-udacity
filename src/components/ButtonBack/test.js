import React from 'react';
import { shallow } from 'enzyme';
import ButtonBack from './index';

describe('[Component] ButtonBack', () => {
  
    it('Display Button Back Component', () => {
        const wrapper = shallow(<ButtonBack />);
        expect(wrapper).toMatchSnapshot();
    });
    
});