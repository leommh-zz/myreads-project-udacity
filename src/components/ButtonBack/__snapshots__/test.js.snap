// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] ButtonBack Display Button Back Component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonBack />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Link
        replace={false}
        to="/"
>
        <WithStyles(Button)
                aria-label="Close"
                style={
                        Object {
                                "backgroundColor": "#c8a3ff",
                                "marginBottom": 10,
                              }
                }
                variant="fab"
        >
                <pure(ArrowBackIcon) />
        </WithStyles(Button)>
</Link>,
      "item": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Button)
          aria-label="Close"
          style={
                    Object {
                              "backgroundColor": "#c8a3ff",
                              "marginBottom": 10,
                            }
          }
          variant="fab"
>
          <pure(ArrowBackIcon) />
</WithStyles(Button)>,
        "replace": false,
        "to": "/",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "aria-label": "Close",
          "children": <pure(ArrowBackIcon) />,
          "style": Object {
            "backgroundColor": "#c8a3ff",
            "marginBottom": 10,
          },
          "variant": "fab",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Link
          replace={false}
          to="/"
>
          <WithStyles(Button)
                    aria-label="Close"
                    style={
                              Object {
                                        "backgroundColor": "#c8a3ff",
                                        "marginBottom": 10,
                                      }
                    }
                    variant="fab"
          >
                    <pure(ArrowBackIcon) />
          </WithStyles(Button)>
</Link>,
        "item": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Button)
            aria-label="Close"
            style={
                        Object {
                                    "backgroundColor": "#c8a3ff",
                                    "marginBottom": 10,
                                  }
            }
            variant="fab"
>
            <pure(ArrowBackIcon) />
</WithStyles(Button)>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "aria-label": "Close",
            "children": <pure(ArrowBackIcon) />,
            "style": Object {
              "backgroundColor": "#c8a3ff",
              "marginBottom": 10,
            },
            "variant": "fab",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
